File: task_equivalents/llm_translations/starcoder/base/r/machine_learning_workflows/02_regression_workflow_feature_importance.R
library(MASS)
library(tidyverse)
library(caret)
Hitters <- read_csv('data/csv/Hitters.csv') %>% drop_na()
Hitters <- Hitters %>% select(-rownames, -Player, -Name)
Hitters <- Hitters %>% mutate(League = ifelse(League == "A", 0, 1), NewLeague = ifelse(NewLeague == "A", 0, 1), Division = ifelse(Division == "E", 0, 1))
Hitters <- Hitters %>% mutate(logSal = log(Salary))

x <- Hitters %>% select(-Salary, -logSal)
y <- Hitters$logSal
train_idx <- createDataPartition(y, p=200/nrow(Hitters), list=FALSE)
xtrain <- x[train_idx, ]
ytrain <- y[train_idx]
xtest <- x[-train_idx, ]
ytest <- y[-train_idx]
lambda_grid <- seq(0.001, 0.201, length.out=20)
train_mse <- rep(0, 20)
for (i in 1:20) {
  boost <- train(xtrain, ytrain, method="gbm", trControl=trainControl(method="none"), tuneGrid=data.frame(n.trees=1000, interaction.depth=1, shrinkage=lambda_grid[i], n.minobsinnode=10))
  train_mse[i] <- mean((ytrain - predict(boost, xtrain))^2)
}
lasso <- train(xtrain, ytrain, method="lasso", trControl=trainControl(method="cv", number=10))
lasso_mse <- mean((ytest - predict(lasso, xtest))^2)
best_lambda <- lambda_grid[which.min(train_mse)]
best_boost <- train(xtrain, ytrain, method="gbm", trControl=trainControl(method="none"), tuneGrid=data.frame(n.trees=1000, interaction.depth=1, shrinkage=best_lambda, n.minobsinnode=10))
feature_importance <- varImp(best_boost)$importance %>% rownames_to_column("Variable") %>% arrange(desc(Overall))
print(feature_importance)

File: task_equivalents/llm_translations/starcoder/base/python/machine_learning_workflows/02_regression_workflow_feature_importance.py
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error
from sklearn.ensemble import GradientBoostingRegressor
from sklearn.linear_model import LassoCV, Lasso
Hitters = pd.read_csv('data/csv/Hitters.csv', header=0, index_col=0)
Hitters = Hitters.dropna()
Hitters = Hitters.drop(['Player', 'Name'], axis=1)
Hitters['League'] = (Hitters['League'] == 'A').astype(int)
Hitters['NewLeague'] = (Hitters['NewLeague'] == 'A').astype(int)
Hitters['Division'] = (Hitters['Division'] == 'E').astype(int)
Hitters['logSal'] = np.log(Hitters['Salary'])
x = Hitters.drop(['Salary', 'logSal'], axis=1)
y = Hitters['logSal']
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.25, random_state=1)
train_mse = []
lambda_grid = np.linspace(0.001, 0.201, 20)
for lambda_val in lambda_grid:
    boost_hit = GradientBoostingRegressor(n_estimators=1000, learning_rate=lambda_val, random_state=1)
    boost_hit.fit(x_train, y_train)
    boost_pred = boost_hit.predict(x_train)
    train_mse.append(mean_squared_error(y_train, boost_pred))
lasso_cv = LassoCV(alphas=lambda_grid, cv=10, random_state=1)
lasso_cv.fit(x_train, y_train)
best_lambda = lasso_cv.alpha_
lasso = Lasso(alpha=best_lambda, random_state=1)
lasso.fit(x_train, y_train)
lasso_pred = lasso.predict(x_test)
lasso_mse = mean_squared_error(y_test, lasso_pred)